maestro-studio/web/src/ReplView.tsx

```diff
@@ -0,0 +1,352 @@
+      <path fillRule="evenodd" d="M5.47 5.47a.75.75 0 011.06 0L12 10.94l5.47-5.47a.75.75 0 111.06 1.06L13.06 12l5.47 5.47a.75.75 0 11-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 01-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 010-1.06z" clipRule="evenodd" />
+    </svg>
+  )
+}
+
+const LoadingIcon = () => {
+  return (
+    <svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" className="w-5 h-5 animate-spin">
+      <path d="M23.9997 11.9187C24.0446 18.546 18.7085 23.9548 12.0812 23.9997C5.45398 24.0446 0.0451292 18.7085 0.000234429 12.0813C-0.0446603 5.454 5.29141 0.0451486 11.9187 0.000253852C18.5459 -0.0446409 23.9548 5.29142 23.9997 11.9187ZM3.20567 12.0596C3.23857 16.9165 7.20258 20.8272 12.0595 20.7943C16.9165 20.7614 20.8272 16.7974 20.7942 11.9404C20.7613 7.08345 16.7973 3.17279 11.9404 3.20569C7.08343 3.23859 3.17277 7.2026 3.20567 12.0596Z" fill="#2563EB" fillOpacity="0.4"/>
+      <path d="M23.9997 11.9187C24.0168 14.4512 23.2323 16.9242 21.7584 18.9837C20.2846 21.0432 18.1969 22.5836 15.7942 23.3843C13.3916 24.1851 10.7972 24.2052 8.38251 23.4417C5.96776 22.6783 3.85647 21.1704 2.35085 19.134L4.92838 17.2283C6.0318 18.7208 7.57912 19.8258 9.34882 20.3854C11.1185 20.9449 13.0199 20.9301 14.7807 20.3433C16.5415 19.7564 18.0715 18.6275 19.1517 17.1182C20.2319 15.6088 20.8068 13.7964 20.7942 11.9404L23.9997 11.9187Z" fill="#2563EB"/>
+    </svg>
+  )
+}
+
+const CheckBox = ({type, checked, onChange}: {
+  type: 'circle' | 'square'
+  checked: boolean
+  onChange?: (checked: boolean) => void
+}) => {
+  const onClick = () => {
+    onChange && onChange(!checked)
+  }
+  if (type === 'circle') {
+    if (checked) {
+      return (
+        <svg
+          viewBox="0 0 20 20"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+          className="w-5 h-5"
+          onClick={onClick}
+        >
+          <rect width="20" height="20" rx="10" fill="#2563EB"/>
+          <path d="M7 10.25L9.6 12.85L13.5 7" stroke="white" strokeLinecap="round" strokeLinejoin="round"/>
+        </svg>
+      )
+    } else {
+      return (
+        <svg
+          viewBox="0 0 20 20"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+          className="w-5 h-5"
+          onClick={onClick}
+        >
+          <rect x="0.5" y="0.5" width="19" height="19" rx="9.5" stroke="#CBD5E1"/>
+        </svg>
+      );
+    }
+  } else {
+    if (checked) {
+      return (
+        <svg
+          viewBox="0 0 20 21"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+          className="w-5 h-5 text-slate-900 hover:text-slate-600 active:text-slate-400"
+          onClick={onClick}
+        >
+          <rect width="20" height="20" transform="translate(0 0.5)" fill="currentColor"/>
+          <path d="M7 10.75L9.6 13.35L13.5 7.5" stroke="white" strokeLinecap="round" strokeLinejoin="round"/>
+        </svg>
+      )
+    } else {
+      return (
+        <svg
```

Comment:
There are a few icons in Studio now. Can we break them all out into a single component that renders the right icon based on a prop like we do internally? i.e. `<Icon id="<id>" />`

Reply:
These icons are more single-use and are scoped to the file that uses them. I don't know if the proposed change would be an improvement tbh

Reply:
Let's revisit if we find ourselves wanting to reuse any of these icons